### ISSUE TITLE
PixelSNAIL overfitting issue

### ISSUE THREAD

**Issue by vipul109** [rosinality/vq-vae-2-pytorch#66]
Hi , First of all thanks for the implementation.

I have tried to train PixelSNAIL-bottom/top prior for 256(imagenet) and 512(gaming) resolution images but I found that both the models are causing overfitting issue .

Bottom-prior (Average train accuracy = 0.77 , validation accuracy: 0.67, test accuracy: 0.37), where train and validation split(9:1) from same datset of 5k images of 512*512 , testing data is another dataset of same class.

Top-prior (Average train accuracy = 0.97180 , validation: 0.88 , testing accuracy: 0.4 ) rest of the settings are same as bottom prior.

I have tried to use l2 regularization, augmentation dataset along with existing dropout but no success. Any lead would be helpful.
Thanks in advance.

**Comment by rosinality** [comment]
Large gap between validation and test is somewhat strange. Anyway, I think there will be not many methods that can be applied to reduce such large overfitting.

**Comment by vipul109** [comment]
Rosinality, Thanks for the reply.
Could you point me to the public datasets which have worked for you?  Did you experiment with FFHQ dataset?

**Comment by rosinality** [comment]
I have tried to train on FFHQ.

**Comment by vipul109** [comment]
Rosinality,
For FFHQ, Did you follow the setting mentioned in the paper ? What is the best accuracy you  achieved?

**Comment by rosinality** [comment]
No, model in the paper is too large to use in my environments. In my cases I got 45% training accuracies for top level codes.

**Comment by wwlCape** [comment]
Hi，Thank you for your VQ-vae2 PyTorch version!
If I want to achieve the results in the paper, what should I do? Just change the holistic architecture bottom+top to bottom+middle+top? Thank you for your reply!

**Comment by rosinality** [comment]
@wwlCape If you want to try 1024 model then you need to use bottom + middle + top models, and larger pixelsnail model. But I don't know this repository can replicate the results in the paper.

**Comment by wwlCape** [comment]
OK, Thanks for your reply!
